cranelift-codegen/src/verifier/mod.rs

```diff
@@ -468,6 +471,57 @@ impl<'a> Verifier<'a> {
         Ok(())
     }
 
+    /// Check that the given EBB can be encoded as a BB, by checking that only
+    /// branching instructions are ending the EBB.
+    fn encodable_as_bb(&self, ebb: Ebb, errors: &mut VerifierErrors) -> VerifierStepResult<()> {
+        // Skip this verification if the environment variable is not set.
+        if !self.verify_encodable_as_bb {
+            return Ok(());
+        };
+
+        let dfg = &self.func.dfg;
+        let inst_iter = self.func.layout.ebb_insts(ebb);
+        // Skip non-branching instructions.
+        let mut inst_iter = inst_iter.skip_while(|&inst| !dfg[inst].opcode().is_branch());
+        let br = inst_iter.next();
+        let br_1 = inst_iter.next();
+        let br_2 = inst_iter.next();
+        match (br, br_1, br_2) {
+            // Terminator instruction is not a branch.
+            (None, _, _) => Ok(()),
+            // Terminator instruction is a branch, but nothing else.
+            (Some(_), None, _) => Ok(()),
+            (Some(_), Some(end), None) => {
+                // Allow 2 consecutive branches at the end of an ebb, unless the
+                // second instruction is a br_table or it is returning from the
+                // current function.
+                if dfg[end].opcode() == Opcode::BrTable {
+                    fatal!(
+                        errors,
+                        end,
+                        "br_table instruction found after a branching instructions in {}",
+                        ebb
+                    )
+                } else if dfg[end].opcode().is_return() {
+                    fatal!(
+                        errors,
+                        end,
+                        "return instruction found after a branching instructions in {}",
+                        ebb
+                    )
+                } else {
+                    Ok(())
+                }
+            }
+            (Some(_), Some(inst), Some(_)) => fatal!(
```

Comment:
Yes, but here we only know that the first element of the tuple is a branch instruction. The fact of having any instruction after a branch which is not a branching terminator is enough. The second instruction is not a teminator, as it would not have more instructions after it.